server/src/main/java/org/elasticsearch/index/mapper/FieldDataParseHelper.java

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefIterator;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.util.ByteUtils;
+import org.elasticsearch.core.CheckedFunction;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * Helper class for processing field data of any type, as provided by the {@link XContentParser}.
+ */
+public final class FieldDataParseHelper {
```

Comment:
I don't think moving the code from `IgnoreMalformedStoredValues` breaks bwc. 
I'm quickly scanning for rest tests that are also run in mixed cluster tests that use ignore malformed feature, but I don't see it. Can we make sure this bwc tested?

Reply:
Hm how is bwc involved here? 

I think we need the node feature since (a) we add a new metadata field, (b) we add tests for ignored fields that won't work without it. Is this what you're referring to?

Reply:
I'm referring to mappings with synthetic source enabled and with boolean, number and ip fields that ignores malformed values. In these cases the code that moved from `IgnoreMalformedStoredValues` to this class is also used.  

But after looking again I did find `ip with ignore_malformed` in `100_synthetic_source.yml` file, but I didn't see any yet for boolean and number fields. The `100_synthetic_source.yml` is part of tests in rest-api-spec module and is run as part of mixed cluster qa module.

Reply:
I think the behavior for ignore_malformed stays unaffected - IgnoreMalformedStoredValues is still used with the same logic that's refactored to a separate class. There should be no change breaking bwc here, afaict. Or, are you just wondering if we have testing coverage to confirm this?

Reply:
> Or, are you just wondering if we have testing coverage to confirm this?

Yes, and it looks like we only have tests for `ip` field type. So maybe we can add tests for number and boolean field types too in `100_synthetic_source.yml`?

Reply:
btw if you're going to add tests maybe add it via another pr. To avoid this one from growing.

Reply:
Right https://github.com/elastic/elasticsearch/pull/107769/ incoming.